### DISCUSSION TITLE
Serializable com.schibsted.spt.data.jslt.Expression

### DISCUSSION THREAD

**Discussion by AlexaBalu** [schibsted/jslt#300] [Ideas]
I found that useful if it comes to spread transformation across nodes. In my use case this is Apache Beam. I compile the Expression in a constructor of DoFn but it fails during the serialization during spreading across all the computational nodes. I'm not sure if expression itself is already bound to something heavy which can't be serialized. Looking at the code this is just an interpretation of JSLT expression.

**Comment by larsga** [comment]
Have you considered spreading the string instead and reparsing it on each node?

That should be much more efficient, as the string will be much more compact than serialized Java objects. The expression refers to `Function` objects, which may be supplied by the implementor, so there is a knock-on effect into the external API here.

Overall, I suspect spreading strings is probably better in pretty much every way.
